pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -220,6 +225,39 @@ func sortedKeys(m map[string]string) []string {
 	return keys
 }
 
+func (p *LaunchTemplateProvider) getBottleRocketUserData(ctx context.Context, constraints *v1alpha1.Constraints, additionalLabels map[string]string) (string, error) {
+	var userData bytes.Buffer
+	userData.WriteString(fmt.Sprintf(`[settings.kubernetes]
+cluster-name = "%s"
+api-server = "%s"
+`,
+		injection.GetOptions(ctx).ClusterName,
+		injection.GetOptions(ctx).ClusterEndpoint))
```

Comment:
We might need to also handle `cluster-dns-ip` now that we support it in the [provisioner](https://karpenter.sh/docs/provisioner/#speckubeletconfiguration)